arangod/VocBase/vocbase.cpp

```diff
@@ -1372,6 +1458,55 @@ arangodb::Result TRI_vocbase_t::dropCollection(DataSourceId cid,
   }
 }
 
+arangodb::Result TRI_vocbase_t::validateCollectionParameters(
+    arangodb::velocypack::Slice parameters) {
+  bool valid = parameters.isObject();
+  if (valid) {
+    // check that the name does not contain any strange characters
+    std::string name = VelocyPackHelper::getStringValue(
+        parameters, StaticStrings::DataSourceName, "");
+    bool isSystem = VelocyPackHelper::getBooleanValue(
+        parameters, StaticStrings::DataSourceSystem, false);
+    bool extendedNames =
+        server().getFeature<DatabaseFeature>().extendedNamesForCollections();
+    valid =
+        CollectionNameValidator::isAllowedName(isSystem, extendedNames, name);
+  }
+
+  if (!valid) {
+    return {TRI_ERROR_ARANGO_ILLEGAL_NAME};
+  }
+
+  TRI_col_type_e collectionType =
+      VelocyPackHelper::getNumericValue<TRI_col_type_e, int>(
+          parameters, StaticStrings::DataSourceType, TRI_COL_TYPE_DOCUMENT);
+
+  if (collectionType != TRI_col_type_e::TRI_COL_TYPE_DOCUMENT &&
+      collectionType != TRI_col_type_e::TRI_COL_TYPE_EDGE) {
+    return {TRI_ERROR_ARANGO_COLLECTION_TYPE_INVALID};
```

Comment:
See above, can we include some context, e.g. the name of the collection?

Reply:
Done